### DISCUSSION TITLE
Is there a way to install Kong Enterprise using helm operator

### DISCUSSION THREAD

**Discussion by morningspace** [Kong/kong#9443] [Help] [answered]
Hi, our team has been used Kong OSS for a few years in our product based on Kubernetes, and now is evaluating Kong Enterprise. I know that for OSS, there's helm operator which can be used to deploy Kong along w/ Kong Ingress Controller to Kubernetes. What about Kong Enterprise?

By looking at the documentation at https://docs.konghq.com/gateway/latest/install/, I can see there's instructions on Kubernetes deployment using `YAML (Manifests)` or `Helm`. However, it looks only `Helm` is `Production Ready`, and there's no such support for Helm deploy using operator. If that's the case, does it mean we need to develop our own operator to support this, or extend the existing one: https://github.com/Kong/kong-operator w/ modification such as using image for Enterprise edition instead, allowing license secret to be specified, etc.

P.S.: I see the deprecation notice on README at: https://github.com/Kong/kong-operator, and it said there's a go-based on w/ the link added in the notice. However the link is broken w/ 404. Not sure if it's not existing or something internally yet.

**Comment by robincher** [comment]
Hi @morningspace,

The new Kong Operator is currently being revamped with the new Kubernetes Gateway API , and its currently in incubation (https://incubator.konghq.com/p/gateway-operator/)

I am based in Asia-Pacific, let me know how i can help, else i can refer you to my colleagues that based in a region that can better correspond with you.

**Comment by hbagdi** [comment]
cc @Kong/team-k8s

**Comment by rainest** [comment, marked as answer]
As https://github.com/Kong/kong-operator is effectively a wrapper around the Helm chart, its `Kong` CRD supports anything the chart values.yaml does. The chart version used in that operator is quite old, but you can use the Enterprise instructions/examples from that version:

https://github.com/Kong/charts/tree/kong-2.4.0/charts/kong#kong-enterprise
https://github.com/Kong/charts/tree/kong-2.4.0/charts/kong/example-values

We do not recommend using that operator, however, as we have no plans to update it further. There may be compatibility issues with that old chart version and current versions of Kong/KIC. We recommend simply using Helm directly, but you could use the Operator SDK to create and maintain your own Helm-based operator if you wish.

**Comment by morningspace** [comment]
Thanks @rainest and @robincher, that information helps us a lot!
